Update sass.js

Spacexplorer11 opened the pull request:
Inconsistencies
On line 59, it says to make a my_theme.scss file, but later at line 81 it tells you to import my-theme.css which doesn't exist
I'm not sure if this was intentional but it caused confusion for me so I thought it'd be best to report it and provide a fix! Thanks!

Spacexplorer11 committed:
Update sass.js

Inconsistencies
On line 59, it says to make a my_theme.scss file, but later at line 81 it tells you to import my-theme.css which doesn't exist

sporeball commented:
when you make `assets/my-theme.scss`, Jekyll converts it from SCSS to CSS when the site is built. if you look in the `_site` folder, you'll find `assets/my-theme.css` instead - so, the Sass task is correct.

i agree that this could be clarified, but i won't merge this PR as-is.

Spacexplorer11 commented:
Yes but there is another issue, `/assets/my-theme.scss'` the forward slash before assets caused issues for me. I'll add another commit, maybe to clarify it

Spacexplorer11 committed:
Update sass.js

Remove the forward slash

Spacexplorer11 commented:
Also, the head.html thing is importing my-theme.scss file which we made, so the Liquid stuff should say `scss` since that is what being imported, then when jekyll serves it, it becomes CSS, if I am wrong lemme know and I'll add a commit to revert my changes and clarify instead

sporeball commented:
what is the specific issue you experienced with the forward slash before `assets`?

the whole line in question was taken [directly from lifeblood](https://github.com/sporeball/lifeblood/blob/master/_includes/head.html#L34), which has had no issues with this for the better part of four years.

Spacexplorer11 commented:
My structure is 
<img width="312" alt="image" src="https://github.com/user-attachments/assets/29006b53-ea21-4261-9bda-083c6438d1ab" />
And this was the error I got when I used `/assets/roundedRectangles.scss` 
<img width="473" alt="image" src="https://github.com/user-attachments/assets/356b26a7-3f74-4f08-9eb3-ef30b805b127" />
I dunno if that was me putting in the wrong place, but from what I understood that was where I was supposed to put it, so either more clarification or there is an error.

sporeball commented:
your structure is correct, but remember: the code that you write in `_includes/head.html` needs to reference your stylesheet *as it will appear in the `_site` folder*. referencing the `.scss` file, [as your theme currently does](https://github.com/Spacexplorer11/RoundedRectangles/blob/6f91d163f044134652f1aa6410333c01998328c0/_includes/head.html#L6), will not work, because of the reason i stated.

when you use the command `bundle exec jekyll serve`, the only thing your web browser sees is the `_site` folder, which does not include the SCSS version of your styles.

Spacexplorer11 commented:
So lets clarify. Is it supposed to be `{{ 'assets/roundedRectangles.scss' | relative_url }}`, `{{ 'assets/roundedRectangles.css' | relative_url }}` ( but keep the file `scss` or change to `css` ),  {{ '/assets/roundedRectangles.scss' | relative_url }}`  or `{{ '/assets/roundedRectangles.css' | relative_url }}`

sporeball commented:
it should be `{{ '/assets/roundedRectangles.css' | relative_url }}`. no changes should be made to your directory structure.

Spacexplorer11 commented:
OK! That worked, Idk what went wrong previously, I'll just add a little bit of clarification

Spacexplorer11 committed:
Update sass.js

Added clarification to why it is ".css" in the code!

Spacexplorer11 commented:
If you find it OK, please merge the clarification!

Spacexplorer11 committed:
Update jekyll_setup.js

Told the user to run `bundle install` which many users have found an issue with as the installation wasn't completed.

Spacexplorer11 commented:
I have fixed the issue where users had to run `bundle install` separately. I have added that to the script. The wording may not be perfect so lemme know if you wanna change it, or if it's good, please merge! Hit 2 birds with one stone in this PR!

Spacexplorer11 committed:
Merge branch 'hackclub:main' into main
